mesheryctl/internal/cli/root/perf/perf.go

```diff
@@ -129,79 +127,75 @@ var PerfCmd = &cobra.Command{
 	},
 	RunE: func(cmd *cobra.Command, args []string) error {
 		// Importing SMPS Configuration from the file
+		var req *http.Request
 		if filePath != "" {
-			var t models.PerformanceSpec
-			err := yaml.Unmarshal([]byte(filePath), &t)
+			smpsConfig, err := ioutil.ReadFile(filePath)
+
 			if err != nil {
-				return errors.Wrapf(err, utils.PerfError(fmt.Sprintf("failed to unmarshal yaml file %s", filePath)))
-			}
-			if testDuration == "" {
-				testDuration = fmt.Sprintf("%fs", t.EndTime.Sub(t.StartTime).Seconds())
+				return err
 			}
-			if testURL == "" {
-				testURL = t.EndpointURL
-			}
-			if concurrentRequests == "" {
-				concurrentRequests = fmt.Sprintf("%d", t.Client.Connections)
+
+			req, err = http.NewRequest("POST", mctlCfg.GetPerf().LoadTestSmpsURI, bytes.NewBuffer(smpsConfig))
+			if err != nil {
+				return errors.Wrapf(err, utils.PerfError(fmt.Sprintf("failed to create new request to %s", mctlCfg.GetBaseMesheryURL())))
 			}
-			if qps == "" {
-				qps = fmt.Sprintf("%f", t.Client.Rps)
+		} else {
+
+			if testName == "" {
+				log.Debug("Test Name not provided")
+				testName = StringWithCharset(8)
+				log.Debug("Using random test name: ", testName)
 			}
-		}
 
-		if testName == "" {
-			log.Debug("Test Name not provided")
-			testName = StringWithCharset(8)
-			log.Debug("Using random test name: ", testName)
-		}
+			postData := ""
 
-		postData := ""
+			startTime := time.Now()
+			duration, err := time.ParseDuration(testDuration)
+			if err != nil {
+				return errors.Wrapf(err, utils.PerfError(fmt.Sprintf("failed to parse test duration %s", testDuration)))
+			}
 
-		startTime := time.Now()
-		duration, err := time.ParseDuration(testDuration)
-		if err != nil {
-			return errors.Wrapf(err, utils.PerfError(fmt.Sprintf("failed to parse test duration %s", testDuration)))
-		}
+			endTime := startTime.Add(duration)
 
-		endTime := startTime.Add(duration)
+			postData = postData + "start_time: " + startTime.Format(time.RFC3339)
+			postData = postData + "\nend_time: " + endTime.Format(time.RFC3339)
 
-		postData = postData + "start_time: " + startTime.Format(time.RFC3339)
-		postData = postData + "\nend_time: " + endTime.Format(time.RFC3339)
+			if testURL != "" {
+				postData = postData + "\nendpoint_url: " + testURL
+			} else {
+				return errors.New(utils.PerfError("please enter a test URL"))
+			}
 
-		if testURL != "" {
-			postData = postData + "\nendpoint_url: " + testURL
-		} else {
-			return errors.New(utils.PerfError("please enter a test URL"))
-		}
+			// Method to check if the entered Test URL is valid or not
+			var validURL = govalidator.IsURL(testURL)
 
-		// Method to check if the entered Test URL is valid or not
-		var validURL = govalidator.IsURL(testURL)
+			if !validURL {
+				return errors.New(utils.PerfError("please enter a valid test URL"))
```

Comment:
@kanishkarj Can we also log the errors using logrus ?

Reply:
I think error logging is handled separately... I am not sure who, But someone who restructured the commands to return errors instead of logging them?

Reply:
@nitishm maybe?

Reply:
> @nitishm maybe?

Yup I switched to cobra's RunE function.

Reply:
I don't exactly know how cobra's RunE fn works, @nitishm If I log it separately will it get redundant?